core/templates/domain/topic/subtopic.model.ts

```diff
@@ -17,18 +17,19 @@
  * subtopic domain objects.
  */
 
-import { ShortSkillSummary } from
-  'domain/skill/short-skill-summary.model';
+import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
 
 import constants from 'assets/constants';
 
 export interface SubtopicBackendDict {
   'id': number;
   'title': string;
   'skill_ids': string[];
-  'thumbnail_filename': string;
-  'thumbnail_bg_color': string;
-  'url_fragment': string;
+  // These values are passed as null initially, See function
+  // 'createFromTitle' in this file.
```

Comment:
How about simple explanation of what null might represent here? Maybe this can help, try thinking about why null is initially passed. Is it a placeholder or does it mean something else. What do we interpret when we see thumbnail_filename as null.

For example: 
Let's assume that null means that the file hasn't been set yet. So you can write above that a null value for thumbnail_filename indicates that a thumbnail hasn't been uploaded yet for this sub_topic.

+cc @seanlip @vojtechjelinek Do you agree with this pattern here and in other places we use null. Given that the codebase is quite big, we can benefit from a nice description of values. (IIRC, Sean was also requesting for something along this line).

Reply:
Yes, any allowance of null should come with a clear explanation of what "null" means.

Typically we use it for properties that are truly optional. E.g. if it is legitimately possible for a created lesson to not have a thumbnail, then null would be a value that indicates that this lesson doesn't have a thumbnail and that that is intended. But if lessons must always have thumbnail images (and this is e.g. enforced in the creation process -- e.g. before creating a new skill you must provide an image), then null should not be an allowed value.

Reply:
Yeah, agreed.

Reply:
Updated, please take a look!